app/controllers/application_controller.rb

```diff
@@ -31,6 +31,13 @@ def authorize_user!
     return false
   end
 
+  def authorize_instructor!
+    return true if is_an_instructor? || has_api_token?
```

Comment:
That uses `submissions#index`, which is exempted from the `authorize_instructor!` call.